### ISSUE TITLE
Anchor Link - No Scoll / Disable Scroll too Anchor/ID option

### ISSUE THREAD

**Issue by ElectricGrape** [elementor/elementor#8808]
## Prerequisites

- [x] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [ ] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

**What problem is your feature request going to solve? Please describe.**

Anchor Links do not have a "disable scroll-to" option. 

**Describe the solution you'd like**

Elementor appears to handle auto-scrolling when using any anchor links, please allow option in anchor attributes, to disable scrolling when clicked by a user. 

Clicking an anchor link will take you directly to that point on the page, no scrolling or other transitions occur. It the UX will be equal to clicking on a link to a new page. 

**Describe alternatives you've considered**

Playing with the code or CSS options but it's not clear anyone else has provided a solution to address exactly this issue, and it seems to me that it's best handled by being an active option within elementor itself, be it free or pro, since it may be a script handled, so some do not want to go there. 

**Additional context**

In UX and design terms, a no scroll go to anchor would be great for some or all anchors in certain design contexts, types and applications.  

Real world:

Scroll is not always the ideal transition experience from a UX point of view. 

There are times scrolling showcases more as the user transitions through a page, like sitting on a train and staring out the window, but it can also be visually jarring in some colour and layouts designs and types of site. Think - Motion sickness.

Minimal is not just design, it's UX too. Minimal events, minimal motions etc. etc. increases focus and attention spans.

**Comment by josiahdahl** [comment]
@ElectricGrape  I ran into this exact issue and you can disable the scroll with custom JavaScript. You should only run this code where you know Elementor is active OR add checks to see if Elementor is in the page, otherwise you'll get an error. This code works with Elementor 2.9.12, ymmv

```
$(document).ready(function() {
  elementorFrontend.elements.$document.off(
    'click', 
    elementorFrontend.utils.anchors.getSettings('selectors.links'),
    elementorFrontend.utils.anchors.handleAnchorLinks
  );
});
```

**Explanation**

[`elementorFrontend.utils.anchors` is set here](https://github.com/elementor/elementor/blob/master/assets/dev/js/frontend/frontend.js#L135)

[The event listener is added here](https://github.com/elementor/elementor/blob/master/assets/dev/js/frontend/utils/anchors.js#L23)

All we need to do is remove the event listener from the `document` and scrolling behavior is removed.

I hope that the Elementor team will make this a configurable option either per page or site wide.

**Comment by PaulKsi** [comment]
Easiest solution for me was to create a html empty element with a div inside, wordpress one, so i can see where it is, and put in elementor section, then just id it and thats it, no more issues

**Comment by zevrix** [comment]
@josiahdahl: Thanks for the solution - but I wonder if you can please elaborate on where exactly this JavaScript should be entered? I tried in the theme's header - but it didn't make any difference. Thanks!

> $(document).ready(function() {
>   elementorFrontend.hooks.addAction('init', function () {
>     elementorFrontend.elements.$document.off(
>       'click', 
>       elementorFrontend.utils.anchors.getSettings('selectors.links'),
>       elementorFrontend.utils.anchors.handleAnchorLinks
>     );
>   });
> });

**Comment by josiahdahl** [comment]
@zevrix Do you get any errors during or after the page load? It should work wherever you put it as long as `$` is defined. Maybe try using `jQuery` instead.

**Comment by zevrix** [comment]
@josiahdahl Thanks! I guess my problem is here:

> as long as $ is defined

When it comes to JavaScript I can miss fine details. 

But regardless, I made the recent solution by @PaulKsi work: insert WordPress Custom HTML widget and id a `div` in it. The scroll animation isn't invoked then.

**Comment by webdevs-pro** [comment]
I found a way to change speed
```
elementorFrontend.utils.anchors.setSettings('scrollDuration', 100);
```

**Comment by zevrix** [comment]
@webdevs-pro
> elementorFrontend.utils.anchors.setSettings('scrollDuration', 100);

Thanks! Can you please elaborate: where exactly do you use it?

**Comment by MyNameIsBrian** [comment]
> I found a way to change speed
> 
> ```
> elementorFrontend.utils.anchors.setSettings('scrollDuration', 100);
> ```

Please elaborate! Very curious.

**Comment by webdevs-pro** [comment]
@zevrix @MyNameIsBrian I put it in HTML widget in footer template. After jQuery ready event.

**Comment by eSilverStrike** [comment]
I was actually looking for the opposite. How to add smooth scrolling to posts that also takes into account any sticky menu heights.

My client creates their own blog posts so uses Gutenberg Editor since Elementor is for page layouts. The posts are created in Gutenberg (but wrapped by an Elementor Single Post Template). When you add a Anchor to these posts there is no scrolling. 

Thanks to the info here and the link to the Elementor code for the smooth scrolling I was able to see that I needed to add the css class: **elementor-menu-anchor** to the anchor in the post (as an Additional CSS class). When I did this it enabled the smooth scrolling for the anchor.

Thanks

**Comment by FelixBehr** [comment]
i will leave my solution for the problem here. Please beware! The site i'm working on is complete elementor driven!

```javascript
jQuery(document).ready(() => {
	const interval = setInterval(function() {
		if(elementorFrontend !== undefined && elementorFrontend.utils !== undefined && elementorFrontend.utils.anchors !== undefined){
			elementorFrontend.utils.anchors.setSettings('scrollDuration', 100);
			clearInterval(interval)
		}
	}, 50)
})
```

**Comment by m1ga** [comment]
2024 and we still have to use the workaround to disable the scolling on click. Sadly that prevents `scroll-padding-top`. The workaround still works but I would love to have an option so it won't be enabled.

Please add something in https://github.com/elementor/elementor/blob/4f2417f6b6ee086d404df8a30edf51c3ea4bedd0/assets/dev/js/frontend/utils/anchors.js#L28C2-L28C19 so we don't have to use it all the time

**Comment by m1ga** [comment]
Finally: https://github.com/elementor/elementor/pull/28368 :+1: 
Not tested yet but it looks very promising from a quick code review

**Comment by robstyrrell** [comment]
I have tried both scripts mentioned above, but I'm getting the following error:

Uncaught TypeError: Cannot read properties of undefined (reading 'setSettings')

Does anyone have any suggestions?

**Comment by m1ga** [comment]
@robstyrrell check the PR I've linked. it's using normal CSS/anchor links for quiet some time now, not JS. You don't need to change anything

**Comment by robstyrrell** [comment]
@m1ga Thanks for the response. However, I'm still confused and don't understand how to disable the smooth scrolling for anchor links? I've tried the code given at the bottom of the PR, but it has no effect.

**Comment by m1ga** [comment]
again: CSS. You just unset https://www.w3schools.com/cssref/pr_scroll-behavior.php on the HTML node. So `html {scroll-behavior: unset;}`.

**Comment by robstyrrell** [comment]
@m1ga Yep. That's sorted it. Thanks for your help. Apologies for being a little thick.
